### PR TITLE
[SYCLomatic] Add steps to generate compilation db for setuptools projects

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Signed-off-by: Ahmed, Daiyaan <daiyaan.ahmed@intel.com>

### CODE REVIEW

File: docs/dev_guide/migrate-a-project/generate-compilation-db.rst [hunk 1]
```diff
@@ -115,6 +115,20 @@ in the build directory:
...
+Setuptools is a package development library in Python for facilitating packaging, distribution and installation of Python projects.For projects with setuptools as their build system, the steps to generate a compilation database are as follows:
+
+#. Generate the build log for the project using the verbose flag 
+#. Use ``intercept-build –-parse-build-log`` option to generate the compilation database.
+
+Example:
+
+.. code-block:: none
+
+    python3 setup.py build_ext -v > build.log 2>&1
```
Comment (tomflinda) [inline docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: Are option "-v" and "build_ext" general options for setup.py?

Comment (daiyaan-ahmed6) [reply docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: build_ext is used to build extension module, and -v is the verbose flag. I have mentioned an example command here so it is easy to understand. Do you suggest I should change it?

Comment (daiyaan-ahmed6) [reply docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: Suggest: python3 setup.py <build_option> -v > build.log 2>&1

Comment (daiyaan-ahmed6) [reply docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: Done

**Review by tomflinda — APPROVED** [review]
LGTM

File: docs/dev_guide/migrate-a-project/generate-compilation-db.rst [hunk 2]
```diff
@@ -115,6 +115,20 @@ in the build directory:
 
    CMake 3.10 or higher is recommended as it provides better support for CUDA source code.
 
+Generate a Compilation Database for Setuptools-Based Projects
+--------------------------------------------------------
+Setuptools is a package development library in Python for facilitating packaging, distribution and installation of Python projects.For projects with setuptools as their build system, the steps to generate a compilation database are as follows:
```
Comment (lapel22) [inline docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: Missing space: "of Python projects.For projects" Need space between projects. and For

File: docs/dev_guide/migrate-a-project/generate-compilation-db.rst [hunk 3]
```diff
@@ -115,6 +115,20 @@ in the build directory:
 
    CMake 3.10 or higher is recommended as it provides better support for CUDA source code.
 
+Generate a Compilation Database for Setuptools-Based Projects
+--------------------------------------------------------
+Setuptools is a package development library in Python for facilitating packaging, distribution and installation of Python projects.For projects with setuptools as their build system, the steps to generate a compilation database are as follows:
+
+#. Generate the build log for the project using the verbose flag 
+#. Use ``intercept-build –-parse-build-log`` option to generate the compilation database.
```
Comment (lapel22) [inline docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: Line 122 should have a period at the end to match line 123.

**Review by lapel22 — CHANGES_REQUESTED** [review]
Few small syntax errors (missing space, missing period)

File: docs/dev_guide/migrate-a-project/generate-compilation-db.rst (hunk 2 above)
Comment (daiyaan-ahmed6) [reply docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: Done. Thanks.

File: docs/dev_guide/migrate-a-project/generate-compilation-db.rst (hunk 3 above)
Comment (daiyaan-ahmed6) [reply docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: Done. Thanks.

**Review by lapel22 — APPROVED** [review]
Looks good to me!

File: docs/dev_guide/migrate-a-project/generate-compilation-db.rst [hunk 4]
```diff
@@ -115,6 +115,20 @@ in the build directory:
 
    CMake 3.10 or higher is recommended as it provides better support for CUDA source code.
 
+Generate a Compilation Database for Setuptools-Based Projects
+--------------------------------------------------------
+Setuptools is a package development library in Python for facilitating packaging, distribution and installation of Python projects. For projects with setuptools as their build system, the steps to generate a compilation database are as follows:
+
+#. Generate the build log for the project using the verbose flag.
```
Comment (zhimingwang36) [inline docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: Please be more detail on how to apply the "verbose" flag,  Give example ok is OK.

Comment (daiyaan-ahmed6) [reply docs/dev_guide/migrate-a-project/generate-compilation-db.rst]: I have written it in more detail. Done.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**